easybuild/tools/build_from_specsfile.py

```diff
@@ -0,0 +1,150 @@
+import yaml
+from easybuild.tools.robot import search_easyconfigs
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.base import fancylogger
+
+
+_log = fancylogger.getLogger('specsfile', fname=False)
+
+
+# general specs applicable to all commands
+class Specsfile(object):
+    def __init__(self):
+        self.easybuild_version = None
+        self.robot = False
+        self.software_list = []
+
+    # returns list of all commands - finished
+    def compose_full_paths(self):
+        easyconfigs_full_paths = []
+        for sw in self.software_list:
+            path_to_append = self.get_ec_path(sw)
+
+            if path_to_append is None:
+                continue
+            else:
+                easyconfigs_full_paths.append(path_to_append)
+        return easyconfigs_full_paths
+
+    # single command
+    def get_ec_path(self, sw):
```

Comment:
This function can be removed (and the import for `search_easyconfigs` removed)

Above, you should probably use the `det_full_ec_version` version to determine the "full version" part of the easyconfig filename (the part except the software name and the `.eb` extension).
For the `.eb` extension, use the `EB_FORMAT_EXTENSION` constant.